utils/config-generator.py

```diff
@@ -343,60 +367,26 @@ def import_keys(all_accounts):
     json.dump(public_key_hashs, open(tezdir + "/public_key_hashs", "w"), indent=4)
 
 
-def get_bootstrap_accounts(accounts, keys_list, is_getting_accounts_for_bakers):
-    keys = {}
-    for key in keys_list:
-        key_name = key["name"]
-        bootstrap_balance = accounts[key_name].get("bootstrap_balance", "0")
-        # Don't add accounts with 0 tez to parameters.json
-        if bootstrap_balance == "0":
-            continue
-
-        # If we are handling pubkeys for baker accounts
-        if is_getting_accounts_for_bakers and accounts[key_name].get(
-            "is_bootstrap_baker_account", False
-        ):
-            keys[key_name] = {
-                "key": key["value"]["key"],
-                "bootstrap_balance": bootstrap_balance,
-            }
-        elif (  # We are handling pubkey hashes for regular accounts
-            not is_getting_accounts_for_bakers
-            and not accounts[key_name].get("is_bootstrap_baker_account", True)
-        ):
-            keys[key_name] = {
-                "key": key["value"],
-                "bootstrap_balance": bootstrap_balance,
-            }
-
-    return keys
-
-
-# Get baking account's pubkeys for parameters.json bootstrap_accounts
-def get_bootstrap_baker_accounts_pubkeys(accounts):
-    with open("/var/tezos/client/public_keys", "r") as f:
-        pubkey_list = json.load(f)
-    return get_bootstrap_accounts(
-        accounts, pubkey_list, is_getting_accounts_for_bakers=True
-    )
-
-
-# Get non-baking account's pubkey hashes for parameters.json bootstrap_accounts
-def get_bootstrap_accounts_pubkey_hashes(accounts):
-    with open("/var/tezos/client/public_key_hashs", "r") as f:
-        pubkey_hash_list = json.load(f)
-    return get_bootstrap_accounts(
-        accounts, pubkey_hash_list, is_getting_accounts_for_bakers=False
-    )
-
+#
+# get_genesis_accounts_pubkey_and_balance(accounts) returns a list
+# of lists: [ [key1, balance2], [key2, balance2], ... ] for all of
+# the accounts prepopulated on our new chain.  Currently, if a public
+# key is provided then the account is signed up as a baker from the
+# start.  If just a public key hash is provided, then it is not.  We
+# use a public key if the property "is_bootstrap_baker_account" is
+# either absent or true.
 
 def get_genesis_accounts_pubkey_and_balance(accounts):
     pubkey_and_balance_pairs = []
 
-    for account_values in accounts.values():
-        pubkey_and_balance_pairs.append(
-            [account_values["key"], account_values["bootstrap_balance"]]
-        )
+    for v in accounts.values():
+        if "bootstrap_balance" in v and v["bootstrap_balance"] != "0":
+            key = v.get("pk")
+            if key == None:
+                key = v.get("pkh")
+            if not v.get("is_bootstrap_baker_account", True):
+                key = v.get("pkh")
```

Comment:
This would give an account baking rights if the `is_bootstrap_baker_account` field was not set on the dict. That should not be by default.

Why can't this be simplified to:
```suggestion
        if "bootstrap_balance" in v and v["bootstrap_balance"] != "0":
           if v.get("is_bootstrap_baker_account", False):
               key = v.get("pk")
           else:
                key = v.get("pkh")
```

Reply:
Interestingly, though, it _is_ the default when we create accounts that have not been specified. And, I think that this does make sense because we only really know about baking accounts in the code---we only create accounts for bakers. And extra account that is created, is inserted into `*_values.yaml` by hand.
Now, what we _can_ do is heuristically determine if we believe an account is attached to a baker and set it that way. But until then, I do think that the True default likely makes the most sense. I can look at the heuristic, but it might be a little complicated, not sure.

Reply:
The simplification that you propose is subtly different, I think. My code installs the pkh if the pk doesn't exist even in the bootstrap baker version. That said, this is dealing with a distinction that, perhaps, is not possible in practice as we can't currently specify just a public key hash. We might want to future-proof the code by tossing our toys out of the pram if we only have a pkh when we expect a pk at this stage. We can add that later, if you think that it's a good plan. Otherwise, I'll take your patch as committed in ebc3c33.

Reply:
It is true we only create accounts for bakers. However, it is the default only when bakers specify they have an account and it doesn't exist, and `SHOULD_GENERATE_UNSAFE_DETERMINISTIC_DATA` is true. And the defaulting is happening on the account creation. Not at the point in time where after all the keys have been accounted for, we now get all of the bootstrap account keys and balances. It shouldn't be determined at this point in the code if an account will be given baking rights or not. It should just be getting their key and balance. 

Imagine we are spinning up a new test chain and for one account we forgot to do `is_bootstrap_baker_account: false`. And the owner of the account does not want to be a baker (right away). Their account would still be given baking rights. And this is without gening unsafe data.

Reply:
I think your second comment makes sense as we can't specify a pkh yet. and the patch is good

Reply:
Actually, mkchain also creates the accounts and sets `is_bootstrap_baker_account: true`. That and the filling in of defaults are the only places where our code generates accounts. The heuristic, as I mentioned, is likely superior, but we're pretty close to it as all generated accounts are marked off as true already. So, I think that we're in agreement.